File: cw_monero/ios/Classes/FlutterMethodHandler.swift
import Foundation

struct FlutterMethodHandler {
    let name: String
    let handler: (FlutterMethodCall, @escaping FlutterResult) -> Void
    
    func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
        if name == call.method {
            handler(call, result)
        }
    }
}

File: cw_monero/android/src/main/kotlin/com/cakewallet/monero/FlutterMethodHandler.kt
package com.cakewallet.monero

import io.flutter.plugin.common.MethodCall
import io.flutter.plugin.common.MethodChannel.Result

class FlutterMethodHandler(val name: String, val handler: (MethodCall, Result) -> Unit) {
    fun handle(call: MethodCall, result: Result) {
        if (name == call.method) {
            handler(call, result)
        }
    }
}